
public class Zaehler {
	private Integer einer= new Integer (0);
	private Integer  zehner= new Integer (0);

	public Zaehler(int x, int y) {
		this.einer = x;
		this.zehner = y;
	}

	public void erhoeheUmEins(Zaehler a) throws EinerUeberlauf {
		/*
		 * Soll die „Einer“ um Eins erhöhen, den aktuellen Wert der „Einer“
		 * ausgeben und bei einem Überlauf der „Einer“ die Exception
		 * EinerUeberlauf werfen
		 */
		if (Integer.valueOf(a.einer) >= 9) {
			throw new EinerUeberlauf();
		} else {
			a.einer=Integer.valueOf(a.einer)+1;
			System.out.println("[Zehner|Einer:]" + " " + a.zehner + a.einer);
		
		}

	}

	/*
	 * soll die „Zehner“ um Eins erhöhen, den aktuellen Wert der „Zehner“
	 * ausgeben und bei einem Überlauf der „Zehner“ also Zaehler= 100 die
	 * Exception Ueberlauf werfen.
	 */
	public void erhoeheUmZehn(Zaehler a) throws Ueberlauf {
		if (Integer.valueOf(a.zehner) >= 9) {
			throw new Ueberlauf("Ueberlauf!");
		} else {
			a.zehner=Integer.valueOf(a.zehner)+1;
			System.out.println("[Zehner|Einer:]" + " " + a.zehner + a.einer);
		}
	}

	/*
	 * Soll um Eins hochzählen, indem die Methode erhoeheUmEins aufgerufen wird.
	 * Wird ein Überlauf der „Einer“ durch das Auftreten einer entsprechenden
	 * Exception signalisiert, dann sollen die „Einer“ zurück auf Null gesetzt
	 * und dafür die „Zehner“durch Aufruf von erhoeheUmZehn erhöht werden.
	 * Danach soll dieser Überlauffehler beseitigt sein.
	 */
	public static void zaehlen(Zaehler a) throws Ueberlauf, EinerUeberlauf {
		try{
			a.erhoeheUmEins(a);
		} catch (EinerUeberlauf e) {
			System.err.println("Ein Einerüberlauf fand stat! ");
			a.einer = 0;
			a.erhoeheUmZehn(a);
		}
	}

}
